### DISCUSSION TITLE
What is libproviders.so?

### DISCUSSION THREAD

**Discussion by Yjy20050126** [openssl/openssl#24433] [Q&A] [answered]
What is libproviders.so? the correct run of coprarna need to use this file, but downloads of openssl donot have this file

**Comment by levitte** [comment, marked as answer]
You will have to ask the coprarna folks.  As you say, that's not a library that builds with our source delivers... never has been.

It could be that someone assumes the because we build `libcrypto.so` from stuff in the `crypto/` directory and `libssl.so` from stuff in the `ssl/` directory, we must build `libproviders.so` from stuff in the `providers/` directory.  A search on the web showed me other cases with this same issue, and it seems to have been such assumptions, or a weirdly made configuration file.

I suggest talking with the coprarna folks.

**Reply by Yjy20050126** [reply]
Thank you！I will ask coprarna folks for help.
